### PR TITLE
Fix lighting

### PR DESCRIPTION
**Pull request by sin3point14**
dorime 🙏

### CODE REVIEW

File: rootex/core/renderer/constant_buffer.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "common/common.h"
+#include "core\renderer\shaders\register_locations_pixel_shader.h"
```
Comment (IronicallySerious) [inline rootex/core/renderer/constant_buffer.h:4]: Forward slash

File: rootex/core/renderer/constant_buffer.h [hunk 2]
```diff
@@ -69,13 +70,13 @@ struct LightsInfo
 {
 	Vector3 cameraPos;
 	int pointLightCount = 0;
-	PointLightInfo pointLightInfos[4];
+	PointLightInfo pointLightInfos[MAX_POINT_LIGHTS];
```
Comment (IronicallySerious) [inline rootex/core/renderer/constant_buffer.h:73]: We will later make this a system setting after #247 gets merged, so we can control the number of lights from the editor

File: rootex/core/renderer/shaders/register_locations_pixel_shader.h [hunk 3]
```diff
@@ -7,3 +7,6 @@
 #define PER_OBJECT_PS_HLSL CONCAT(b, PER_OBJECT_PS_CPP)
 #define PER_FRAME_PS_HLSL CONCAT(b, PER_FRAME_PS_CPP)
 #define SKY_PS_HLSL CONCAT(t, SKY_PS_CPP)
+
+#define MAX_POINT_LIGHTS 4
+#define MAX_SPOT_LIGHTS 4
```
Comment (IronicallySerious) [inline rootex/core/renderer/shaders/register_locations_pixel_shader.h:12]: Newline

File: rootex/framework/systems/light_system.cpp [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include "light_system.h"
+#include "core\renderer\shaders\register_locations_pixel_shader.h"
```
Comment (IronicallySerious) [inline rootex/framework/systems/light_system.cpp:2]: Forward slash

**Review by IronicallySerious — CHANGES_REQUESTED** [review]

File: rootex/core/renderer/constant_buffer.h (hunk 2 above)
Comment (sin3point14) [reply rootex/core/renderer/constant_buffer.h:73]: issue is that this info is also used in shaders

**Review by IronicallySerious — APPROVED** [review]

**Pull request merged**